File: 277. Find the Celebrity.py
# The knows API is already defined for you.
# return a bool, whether a knows b
# def knows(a: int, b: int) -> bool:

class Solution:
    def findCelebrity(self, n: int) -> int:
        
        counter = 0
        
        for x in range(n):
            counter = 0
            for y in range(n):
                if knows(y,x) == True and knows(x,y)==False and x!=y:
                    counter+=1
                elif x==y:
                    counter+=1
                else:
                    break
                    
                
                    
            if counter == n:
                return x
        
        return -1
                    
                    
                
        

File: cpp_solutions/277. Find the Celebrity.cpp
/* The knows API is defined for you.
      bool knows(int a, int b); */

class Solution {
public:
    int findCelebrity(int n) {
        //may exist one celebrity
        // everyone knows the celebrity
        // celebrity does not know anyone of them
        
        for(int x=0;x<n;x++){
            int count = 0;
            
            for(int y=0;y<n;y++){
                
                if(x!=y && knows(y,x) && not knows(x,y)){
                    count+=1;
                }
                
                else if(x==y){
                    count+=1;
                }
                else{
                    break;
                }
                
            }
            if (count ==n){
                return x;
            }
        }
        return -1;
        
    }
};
